backend/src/main/java/com/redhat/cloud/notifications/db/repositories/GatewayCertificateRepository.java

```diff
@@ -0,0 +1,71 @@
+package com.redhat.cloud.notifications.db.repositories;
+
+import com.redhat.cloud.notifications.models.Application;
+import com.redhat.cloud.notifications.models.GatewayCertificate;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+import jakarta.persistence.EntityManager;
+import jakarta.transaction.Transactional;
+import jakarta.ws.rs.NotFoundException;
+import java.util.UUID;
+
+@ApplicationScoped
+public class GatewayCertificateRepository {
+
+    @Inject
+    EntityManager entityManager;
+
+    @Inject
+    ApplicationRepository applicationRepository;
+
+    @Transactional
+    public GatewayCertificate createGatewayCertificate(GatewayCertificate gatewayCertificate) {
+        Application application = applicationRepository.getApplication(gatewayCertificate.getBundle(), gatewayCertificate.getApplication());
+        gatewayCertificate.setGatewayCertificateApplication(application);
+        entityManager.persist(gatewayCertificate);
+        return gatewayCertificate;
+    }
+
+
+    public GatewayCertificate findGatewayCertificate(String bundle, String application, String certificateData) {
```

Comment:
We could return an `Optional` (empty if not found) here:
```suggestion
    public Optional<GatewayCertificate> findGatewayCertificate(String bundle, String application, String certificateData) {
```